Scripts/script-DBotPreparePhishingData.yml

```diff
@@ -0,0 +1,219 @@
+  import demisto_ml
+
+  TRAIN_FILE_NAME = str(uuid.uuid4()) + ".train"
+  ALL_LABELS = "*"
+  TAG_PREFIX = demisto_ml.DEFAULT_LABEL_PREFIX
+
+  def get_context(incident_id):
+      res = demisto.executeCommand("getContext", {'id': incident_id});
+      try:
+        return res[0]['Contents']['context']
+      except:
+        return {}
+
+  def canonize_label(label):
+      return label.lower().strip().replace(" ","_")
+
+  def get_phishing_map_labels(comma_values):
+      if comma_values == ALL_LABELS:
+          return comma_values
+      comma_values = comma_values.lower()
+      values = comma_values.split(",")
+      labels_map = {}
+      for v in values:
+          v = canonize_label(v)
+          if ":" in v:
+              splited = v.split(":")
+              labels_map[splited[0].strip()] = canonize_label(splited[1])
+          else:
+              labels_map[v] = canonize_label(v)
+      return labels_map
+
+  def get_incidents(query, size, is_context_data_needed):
+      res = demisto.executeCommand("getIncidents", {"query": query, "size": size, "sort": "created"})
+      if res[0]['Type'] == entryTypes['error']:
+          return_error(str(res[0]['Contents']))
+      incident_list = res[0]['Contents']['data'] or []
+
+      for i in incident_list:
+          custom_fields = i.get('CustomFields', {}) or {}
+          if is_context_data_needed:
+              i.update(get_context(i['id']))
+          i.update(custom_fields)
+
+      return incident_list
+
+  def get_tag(category):
+     return tag.replace(" ","_")
+
+
+  EMAIL_TEXT_DT = demisto.args()['emailTextKey']
+  EMAIL_SUBJECT_DT = demisto.args()['emailSubjectKey']
+  TAG_DT = demisto.args()['tagKey']
+  PHISHING_LABELS = get_phishing_map_labels(demisto.args()['phishingLabels'])
+  IS_CONTEXNT_DATA_NEEDED = (demisto.args()['isContextNeeded'] == 'yes')
+  HASH_WORDS = (demisto.args()['hashData'] == 'yes')
+  LIST_NAME_TO_STORE = demisto.args().get('storeFileInList')
+  HASH_SEED = 5381
+  MAX_INCIDENTS = int(demisto.args()['maxNumberOfIncidents'])
+
+  # get the incidents for training
+  incidents = get_incidents(demisto.args()['incidentsQuery'], MAX_INCIDENTS, IS_CONTEXNT_DATA_NEEDED)
+  # parse the incidents
+  skipped_empty = 0
+  skipped_missing_labels = 0
+  skipped_invalid_label = 0
+  skipped_missing_body = 0
+  skipped_exception = 0
+  labels_value_counts = {}
+  data_texts = []
+  data_labels = []
+  skipped_labels = []
+  if len(incidents) == MAX_INCIDENTS:
+      demisto.log('Number of results returned from incidentsQuery has reached the limit (maxNumberOfIncidents=%d). Consider increase maxNumberOfIncidents to train on more data.' % MAX_INCIDENTS)
+  for i in incidents:
+      try:
+          text = (demisto.dt(i, EMAIL_TEXT_DT) or "")
+          subject = (demisto.dt(i, EMAIL_SUBJECT_DT) or "")
+          if not text and not subject:
+              skipped_empty += 1
+              continue
+          original_label = canonize_label(demisto.dt(i, TAG_DT))
```

Comment:
What does `demisto.dt` returns if it doesn't find the item? (If it's `None` we would have an error in `canonize_label)

Reply:
handled it now